Assets/Scripts/Combat/TargetFinder.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections.Generic;
+
+using UnityEngine;
+
+namespace CwispyStudios.TankMania.Combat
+{
+  public class TargetFinder : MonoBehaviour
+  {
+    [Tooltip("Target must match the Team preference, and must contain ANY of the UnitType selected. If UnitType is 0, accepts anything.")]
+    [SerializeField] private UnitProperties targetPreference;
+
+    private List<Rigidbody> targetsInRange = new List<Rigidbody>();
+    public IList<Rigidbody> TargetsInRange => targetsInRange.AsReadOnly();
+
+    private void OnDisable()
+    {
+      targetsInRange.Clear();
+    }
+
+    private void OnTriggerEnter( Collider other )
```

Comment:
Just realized in experience, OnTriggerEnter and OnTriggerExit don't work well with objects which you despawn or repool, because in some cases the object might die while in the trigger and be relocated, without triggering OnTriggerExit, causing the list to contain objects which are despawned, **should be tested**.

Reply:
Nice catch, it doesn't trigger when an object is disabled

Reply:
Tempory solution - Inherit from a class that invokes an event when disabled